src/test/scala/is/hail/expr/ir/StringSliceSuite.scala

```diff
@@ -0,0 +1,91 @@
+package is.hail.expr.ir
+
+import is.hail.utils._
+import is.hail.TestUtils._
+import is.hail.expr.types._
+import org.scalatest.testng.TestNGSuite
+import org.testng.annotations.Test
+
+class StringSliceSuite extends TestNGSuite {
+  @Test def zeroToLengthIsIdentity() {
+    assertEvalsTo(StringSlice(Str("abc"), I32(0), I32(3)), "abc")
+  }
+
+  @Test def outOfBoundsFatals() {
+    assertFatal(StringSlice(Str("abc"), I32(4), I32(4)),
```

Comment:
Might be worth seeing what Patrick did with the ArraySuite and SetSuites in #3554. He added some nice features.